src/scripts/main.js

```diff
@@ -1,7 +1,307 @@
+    addRandomTile();
+    updateBoard();
+  }
+}
+
+function moveLeft() {
+  let moved = false;
+
+  for (let row = 0; row < 4; row++) {
+    for (let col = 3; col > 0; col--) {
+      if (board[row][col] !== 0) {
+        let nextCol = col - 1;
+
+        while (nextCol >= 0) {
+          if (board[row][nextCol] === 0) {
+            board[row][nextCol] = board[row][col];
+            board[row][col] = 0;
+            col = nextCol;
+            moved = true;
+          } else if (board[row][nextCol] === board[row][col]) {
+            board[row][nextCol] *= 2;
+            score += board[row][nextCol];
+            board[row][col] = 0;
+            moved = true;
+            break;
+          } else {
+            break;
+          }
+          nextCol--;
+        }
+      }
+    }
+  }
+
+  return moved;
+}
+
+function moveRight() {
+  let moved = false;
+
+  for (let row = 0; row < 4; row++) {
+    for (let col = 0; col < 3; col++) {
+      if (board[row][col] !== 0) {
+        let nextCol = col + 1;
+
+        while (nextCol < 4) {
+          if (board[row][nextCol] === 0) {
+            board[row][nextCol] = board[row][col];
+            board[row][col] = 0;
+            col = nextCol - 1;
+            moved = true;
+          } else if (board[row][nextCol] === board[row][col]) {
+            board[row][nextCol] *= 2;
+            score += board[row][nextCol];
+            board[row][col] = 0;
+            moved = true;
+            break;
+          } else {
+            break;
+          }
+          nextCol++;
+        }
+      }
+    }
+  }
+
+  return moved;
+}
+
+function moveUp() {
+  let moved = false;
+
+  for (let col = 0; col < 4; col++) {
+    for (let row = 1; row < 4; row++) {
+      if (board[row][col] !== 0) {
+        let nextRow = row - 1;
+
+        while (nextRow >= 0) {
+          if (board[nextRow][col] === 0) {
+            board[nextRow][col] = board[row][col];
+            board[row][col] = 0;
+            row = nextRow;
+            moved = true;
+          } else if (board[nextRow][col] === board[row][col]) {
+            board[nextRow][col] *= 2;
+            score += board[nextRow][col];
+            board[row][col] = 0;
+            moved = true;
+            break;
+          } else {
+            break;
+          }
+          nextRow--;
+        }
+      }
+    }
+  }
+
+  return moved;
+}
+
+function moveDown() {
+  let moved = false;
+
+  for (let col = 0; col < 4; col++) {
+    for (let row = 2; row >= 0; row--) {
+      if (board[row][col] !== 0) {
+        let nextRow = row + 1;
+
+        while (nextRow < 4) {
+          if (board[nextRow][col] === 0) {
+            board[nextRow][col] = board[row][col];
+            board[row][col] = 0;
+            row = nextRow;
+            moved = true;
+          } else if (board[nextRow][col] === board[row][col]) {
+            board[nextRow][col] *= 2;
+            score += board[nextRow][col];
+            board[row][col] = 0;
+            moved = true;
```

Comment:
maybe you can make your code reusable. In every `move{Arrow}` function you do similar things
For example, create a function `mergeCells` that accepts correct indices (col, row, etc) and changes `board` array, etc